ZAPD/ZString.cpp

```diff
@@ -0,0 +1,58 @@
+#include "ZString.h"
+
+#include "File.h"
+#include "ZFile.h"
+#include "StringHelper.h"
+
+REGISTER_ZFILENODE(String, ZString);
+
+ZString::ZString(ZFile* nParent) : ZResource(nParent)
+{
+}
+
+void ZString::ParseRawData()
+{
+	size_t size = 0;
+    for (size_t i = rawDataIndex; i < rawData.size(); ++i) {
+        ++size;
+        if (rawData.at(i) == '\0') {
```

Comment:
I would recommend getting a `uint8_t*` pointer from rawData.data() and iterating through this looping using that pointer. Calling `at()` or using an array operator on `rawData` tends to have a performance impact if done frequently.

Reply:
I think using `rawData[]` should have the same performance impact as using a `uint8_t*`, but I don't have proof, so I'll do it like you said.

Reply:
When I was doing profiling looking for hotspots in ZAP's code, it was one thing I spotted. I have not looked at the std::vector implementation (and it may vary from compiler to compiler), but I think it might be doing extra work.